lib/std/zig.zig

```diff
@@ -843,6 +844,7 @@ pub const SimpleComptimeReason = enum(u32) {
             .clobber              => "clobber must be comptime-known",
 
             .type                => "types must be comptime-known",
+            .merge_error_set     => "operand to error set merge (||) must be an error set",
```

Comment:
```suggestion
            .merge_error_set     => "operand to error set merge operator (||) must be a type",
```

although, really, it would probably be better to use the `types must be comptime-known` error message, with an additional note explaining *why* it expects a type.